spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1762,15 +1799,21 @@ public Boolean isMandatoryFor(final Message message) {
 		Connection connection = null; // NOSONAR (close)
 		if (channel == null) {
 			if (isChannelTransacted()) {
-				resourceHolder = ConnectionFactoryUtils.getTransactionalResourceHolder(connectionFactory, true);
+				resourceHolder = ConnectionFactoryUtils.
+						getTransactionalResourceHolder(connectionFactory, true, publisherConnectionIfPossible);
 				channel = resourceHolder.getChannel();
 				if (channel == null) {
 					ConnectionFactoryUtils.releaseResources(resourceHolder);
 					throw new IllegalStateException("Resource holder returned a null channel");
 				}
 			}
 			else {
-				connection = connectionFactory.createConnection(); // NOSONAR - RabbitUtils
+				if (publisherConnectionIfPossible) {
+					connection = connectionFactory.createPublisherConnection(); // NOSONAR - RabbitUtils
```

Comment:
I think we don't need all this "PublisherConnection" kung-fu, when we just can obtain a `publisherConnectionFactory` from the provided CF if `usePublisherConnection == true`.
Or am I missing anything else?

Reply:
Again, comment above.